src/bigquery.ts

```diff
@@ -248,12 +250,27 @@ export class BigQuery extends common.Service {
   getJobsStream: (options?: GetJobsOptions) => ResourceStream<Job>;
 
   constructor(options: BigQueryOptions = {}) {
+    let apiEndpoint = 'https://bigquery.googleapis.com';
+
+    const EMULATOR_HOST = process.env.BIGQUERY_EMULATOR_HOST;
+
+    if (typeof EMULATOR_HOST === 'string') {
+      apiEndpoint = BigQuery.sanitizeEndpoint(EMULATOR_HOST);
+    }
+
+    if (options.apiEndpoint) {
+      apiEndpoint = BigQuery.sanitizeEndpoint(options.apiEndpoint);
```

Comment:
Maybe a new test to make sure `options.apiEndpoint` trumps the emulator variable.